Series of small WCOSS2 updates

KateFriedman-NOAA opened the pull request:
Updates included in this PR:

1. WCOSS2.env updates for CFP
2. a couple small waveinit job script updates (runs without failure now)
3. build updates for enkf_chgres_recenter_nc (including conversion to LUA modulefile)
4. change NWROOT to PACKAGEROOT in exglobal_forecast.sh
5. update GSI and WAFS component checkouts

Refs: #399

KateFriedman-NOAA committed:
Update repository name for EMC_post component to UPP in operations branch

- the "EMC_post" repository was renamed to "UPP" on September 6th 2021
- update repository url in checkout script and Externals.cfg

Refs: #433

KateFriedman-NOAA committed:
Update repository name for EMC_post component to UPP in operations branch (#442)

- the "EMC_post" repository was renamed to "UPP" on September 6th 2021
- update repository url in checkout script and Externals.cfg

Refs: #433

KateFriedman-NOAA committed:
Merge branch 'NOAA-EMC:operations' into operations

KateFriedman-NOAA committed:
Update GSI tag to gfsda.v16.1.3 (turn off IASI early) (#449)

- Emergency change occurred in operations during 2021092206 gdas cycle
to turn off the IASI from metop-a earlier than planned
- GSI global_satinfo.txt was updated to turn off IASI
- From SDM log on 20210921: 1ap. 1900Z - IDSB Chief Carissa requested
an ARFC for the SPA Team. The metopA "IASI" instrument is failing,
so they need to make a modification to dump jobs so models do not
assimilate the degraded data. They would like to make the change at
12Z tomorrow. NCO Deputy Director and SDM approved. (MBG)

Refs: #446

KateFriedman-NOAA committed:
Merge branch 'NOAA-EMC:operations' into operations

KateFriedman-NOAA committed:
GFSv16.1.4 - Operational Assimilation of Commercial GPS-RO in DO-3 (#451)

* Update config.anal for historical fix files in GSI

In the v16.1.4 GSI release branch, the gfsv16_historical fix files of
global_convinfo.txt and prepobs_errtable.global have been updated.
New IF blocks in the config.anal file have been added to properly
use these new historical fix files. This allows for proper observation
use in parallels that are not running near realtime.

* Update GSI tag to gfsda.v16.1.4

The GSI tag was updated from gfsda.v16.1.1a to gfsda.v16.1.4.  This tag
turns off GeoOptics assimilation, turns on Spire assimilation, adjusts
the observation errors for the commerical data, corrects two RO
observation operator bugs, and relaxes the gross check for RO
that was added as a mitigation measure for those bugs.

Ref: #420

Co-authored-by: catherine.thomas <catherine.thomas@noaa.gov>

KateFriedman-NOAA committed:
Merge branch 'NOAA-EMC:operations' into operations

KateFriedman-NOAA committed:
GFSv16.1.5 - GSI update to include IASI Metop-C in operations (#439)

RFC 8823 - On WCOSS, update GFS to v16.1.5 which includes:
* Turn on the active assimilation of Metop-C IASI (Infrared Atmospheric Sounding Interferometer) data with correlated observation errors
* Update correlated observation errors for Metop-B IASI, N20 CrIS (Cross-track Infrared Sounder), and NPP CrIS
* Put in a minimum threshold for saturation specific humidity to avoid potential minimization issues.
* Add the 2022 CO2 projection files.

Implemented on October 26 at 1430Z.

KateFriedman-NOAA committed:
LUA and build updates for enkf_chgres_recenter_nc

- create modulefile to build enkf_chgres_recenter_nc on WCOSS2
- convert enkf_chgres_recenter_nc.wcoss2 to LUA
- clean up build_enkf_chgres_recenter_nc.sh
- add -L$(HDF5_LIBRARIES) to enkf_chgres_recenter_nc makefile

Refs: #399

KateFriedman-NOAA committed:
CFP updates in WCOSS2 env file

- adjust launcher commands for jobs that will use CFP/mpmd
- need to use "-np" with CFP instead of just "-n" like non-CFP

Refs: #399

KateFriedman-NOAA committed:
Updates for waveinit job

- change JJOB script to use wavempexec from above instead of launcher to
set the wavempexec variable
- remove duplicate DATA folder cleanup in rocoto job script; it's
handled by the JJOB script as it should

Refs: #399

KateFriedman-NOAA committed:
Component checkout updates for GSI and WAFS

- new GSI branch available (feature/gfsda.v16.1.5_wcoss2_port)
- new WAFS tag available (gfs_wafs.v6.2.0 - pre-cmake)

Refs: #399

KateFriedman-NOAA committed:
Replace NWROOT with PACKAGEROOT forecast driver

- the NWROOT variable is now called PACKAGEROOT on WCOSS2

Refs: #399

KateFriedman-NOAA committed:
Merge branch 'NOAA-EMC:operations' into operations

KateFriedman-NOAA committed:
Sync merge with operations for GFSv16.1.5

- release note updates
- component versions in Externals.cfg updated
- WCOSS_DELL_P3 env update to add "-l" with mpirun
- config.anal updates for historical info file dates

Refs: #399

* origin/operations:
  GFSv16.1.5 - GSI update to include IASI Metop-C in operations (#439)
  GFSv16.1.4 - Operational Assimilation of Commercial GPS-RO in DO-3 (#451)
  Update GSI tag to gfsda.v16.1.3 (turn off IASI early) (#449)
  Update repository name for EMC_post component to UPP in operations branch (#442)
  Update repository name for EMC_post component to UPP in operations branch

KateFriedman-NOAA committed:
Merge branch 'NOAA-EMC:feature/ops-wcoss2' into feature/ops-wcoss2

aerorahul commented on sorc/enkf_chgres_recenter_nc.fd/makefile:
```diff
@@ -1,6 +1,6 @@
 SHELL=  /bin/sh
 
-LIBS= $(FV3GFS_NCIO_LIB) $(BACIO_LIB4) $(W3NCO_LIB4) $(IP_LIB4) $(SP_LIB4) -L$(NETCDF)/lib -lnetcdff -lnetcdf -lhdf5_hl -lhdf5 -lz
+LIBS= $(FV3GFS_NCIO_LIB) $(BACIO_LIB4) $(W3NCO_LIB4) $(IP_LIB4) $(SP_LIB4) -L$(NETCDF)/lib -L$(HDF5_LIBRARIES) -lnetcdff -lnetcdf -lhdf5_hl -lhdf5 -lz
```
```suggestion
LIBS= $(FV3GFS_NCIO_LIB) $(BACIO_LIB4) $(W3NCO_LIB4) $(IP_LIB4) $(SP_LIB4) -L$(HDF5_LIBRARIES) -lhdf5_hl -lhdf5 -lz -L$(NETCDF)/lib  -lnetcdff -lnetcdf 
```

aerorahul reviewed (commented):
looks good. 
Just one comment.

KateFriedman-NOAA committed:
Update sorc/enkf_chgres_recenter_nc.fd/makefile

Co-authored-by: Rahul Mahajan <aerorahul@users.noreply.github.com>

KateFriedman-NOAA committed:
Merge branch 'NOAA-EMC:feature/ops-wcoss2' into feature/ops-wcoss2

KateFriedman-NOAA committed:
Remove duplicate DATA cleanup in waveprep script

- similar to removal in waveinit script, remove duplicate DATA removal
from waveprep rocoto job script

Refs: #399

KateFriedman-NOAA committed:
Add load of udunits module

- add load of udunits/2.2.28 module to module_base.wcoss2.lua
- need to load udunits module with nco module for successful nco utility
usage in waveprep job
- add corresponding udunits module version to run.ver

Refs: #399

KateFriedman-NOAA commented:
@aerorahul I have also included two additional small commits to this PR, which resolve a failing waveprep job (similar to waveinit updates in this PR). Please review and approve if good, thanks!

WalterKolczynski-NOAA commented on sorc/checkout.sh:
```diff
@@ -35,9 +35,9 @@ fi
     cd gsi.fd
-    git checkout gfsda.v16.1.1a
+    #git checkout feature/gfsda.v16.1.5_wcoss2_port
     git submodule update
```
Should this be recursive, just in case another layer of submodules is added?

KateFriedman-NOAA replied:
On line 38 of checkout.sh I am recursively cloning the branch and commented out the secondary branch checkout on line 40 for now.

```
git clone --recursive --branch feature/gfsda.v16.1.5_wcoss2_port https://github.com/NOAA-EMC/GSI.git gsi.fd >> ${topdir}/checkout-gsi.log 2>&1
```

WalterKolczynski-NOAA replied:
That works too. You shouldn't even need to do the submodule update that way.

We should probably pick one way and do all components the same way. Specifying the branch at clone time is probably best.

KateFriedman-NOAA replied:
Agreed. The checkout script is a little messy right now as I set the test branches but I'd like to review and clean it towards the end as we get tags sorted.
